### PR TITLE
input different coordinate systems in initial temperature function

### PR DESCRIPTION
**Pull request by mbweller**
Follow up to boundary condition changes

### CODE REVIEW

**Comment by bangerth** [comment]
Follow-up to #1563.

File: include/aspect/initial_temperature/function.h [hunk 1]
```diff
@@ -78,6 +80,13 @@ namespace aspect
          * A function object representing the temperature.
          */
         Functions::ParsedFunction<dim> function;
+
+        /**
+         * The coordinate representation to evaluate the function. Possible
+         * choices are depth, cartesian and spherical.
+         */
+        ::aspect::Utilities::Coordinates::CoordinateSystem coordinate_system;
```
Comment (bangerth) [inline include/aspect/initial_temperature/function.h]: I don't think you actually need the `::aspect::` at the front. Same everywhere else.

File: source/initial_temperature/function.cc [hunk 2]
```diff
@@ -38,7 +41,33 @@ namespace aspect
...
+          return function.value(position);
+        }
+      else if (coordinate_system == ::aspect::Utilities::Coordinates::CoordinateSystem::spherical)
+        {
+          const std_cxx11::array<double,dim> spherical_coordinates =
+            aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(position);
+          Point<dim> point;
+
+          for (unsigned int i = 0; i<dim; ++i)
+            point[i] = spherical_coordinates[i];
```
Comment (bangerth) [inline source/initial_temperature/function.cc]: see also #1632.

File: source/initial_temperature/function.cc [hunk 3]
```diff
@@ -38,7 +41,33 @@ namespace aspect
...
+
+          return function.value(point);
+        }
+          else if (coordinate_system == ::aspect::Utilities::Coordinates::CoordinateSystem::depth)
+        {
```
Comment (bangerth) [inline source/initial_temperature/function.cc]: wrong indentation here.

File: source/initial_temperature/function.cc [hunk 4]
```diff
@@ -64,6 +111,10 @@ namespace aspect
       prm.enter_subsection ("Initial temperature model");
       {
         prm.enter_subsection("Function");
+        {
+          coordinate_system = ::aspect::Utilities::Coordinates::string_to_coordinate_system(prm.get("Coordinate system"));
```
Comment (bangerth) [inline source/initial_temperature/function.cc]: no need for `::aspect::`

**Comment by mbweller** [comment]
Addressed small issues. Will address functionality in #1632 following this.

**Comment by Shangxin-Liu** [comment]
I'd like to use r, theta, phi to input single degree order temperature field in my geoid benchmark prm file when this get merged.

**Comment by bangerth** [comment]
This looks good to me. Do you think it is practical to add tests for each of these places?

**Comment by Shangxin-Liu** [comment]
Do you mean adding the same test prm file with single degree order perturbation for both geoid and this spherical coordinate initial temperature function?

**Comment by bangerth** [comment]
Ideally, we'd like to have tests that run through every new code path that is added to ASPECT. In this case, that would mean adding tests for each coordinate system for each plugin that you are modifying. That may be a lot of new tests, and I don't know whether that's practical, but it would be nice to have.

**Comment by mbweller** [comment]
Had to make a small change to function.h (broken functionality). It works with test cases.

File: include/aspect/initial_temperature/function.h (hunk 1 above)
Comment (tjhei) [inline include/aspect/initial_temperature/function.h]: It might be needed to differentiate it from ``dealii::Utilities``?

Comment (bangerth) [inline include/aspect/initial_temperature/function.h]: It should find `aspect::Utilities` first when you just say `Utilities`. Can you try, @mbweller ?

Comment (mbweller) [inline include/aspect/initial_temperature/function.h]: I tried, and it didn't work. It didn't give a compilation error, it lost functionality. Specifically it didn't recognize set Coordinate system.

Comment (bangerth) [inline include/aspect/initial_temperature/function.h]: I don't understand this. If it compiles, it must work. Can you post what error message or fault you get?

Comment (bangerth) [inline include/aspect/initial_temperature/function.h]: See #1648.

Comment (mbweller) [inline include/aspect/initial_temperature/function.h]: I can't explain it. I just recompiled, and it works. Earlier it did not, and the only solution was to add ::aspect:: to Utilities:: ...

**Pull request merged**